packages/labs/vue-utils/src/wrapper-utils.ts

```diff
@@ -1,4 +1,41 @@
-import {h, PropType, VNode} from 'vue';
+import {h, PropType, VNode, reactive, Directive} from 'vue';
+
+type PropsDirective<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T = any,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  V = HTMLElement & Record<keyof T, any>
+> = Directive<V, T> & {
+  defaults?: T;
+  setProps: (el: V, value: T, hasRendered?: boolean) => void;
+};
+
+/**
+ * Set properties to bound element, preserving Vue's convention of
+ * applying default values when the bound value is `undefined`. Default values
+ * are recorded at created time, before any properties are set. Then, whenever
+ * the element is updated (including mounted), properties are set to a given
+ * defined value or the default.
+ */
+export const vProps: PropsDirective = {
```

Comment:
We should implement `getSSRProps` to just return the props bag during SSR

Reply:
Done.